Домашнее задание по уроку 2

GareevaFarida opened the pull request.

GareevaFarida committed:
Домашнее задание по уроку 2

usharik commented on src/Application.java:
```diff
@@ -0,0 +1,57 @@
+        int y = Dimentions.Y.getValue();
+
+        if (massive.length != x) {
+            throw new MyArraySizeException("В метод передан массив с несоответствующей размерностью массива. Количество строк в массиве: " + massive.length + " вместо " + x);
```
Лучше передать индексы массива как параметры в исключение и сконструировать сообщение в конструкторе.

usharik commented on src/Application.java:
```diff
@@ -0,0 +1,57 @@
+                        massiveInt[i][j] = Integer.parseInt(massive[i][j]);
+                        summa = summa + massiveInt[i][j];
+                    } catch (NumberFormatException e) {
+                        message_error = message_error + "\n" +
```
Логичнее сразу генерить исключение. Нет смысла продолжать расчет суммы, если один из элементов не корректен.

GareevaFarida replied:
Да, согласна, что высчитывать сумму не логично. Но если массив большой и в нем много ошибок? Пользователь устанет выискивать по одной ошибке, тем более, если нужно передать список ошибок заказчику, чтобы он единоразово все исправил.

usharik replied:
Как правило в таких случаях работу останавливают на первой ошибке, если в задании не указанно обратное;

usharik commented on src/Enums/Dimentions.java:
```diff
@@ -0,0 +1,16 @@
+package Enums;
+
+public enum Dimentions {
```
Конечно так можно, но уж слишком мудрено использовать перечисление для таких значений. Двух констант было бы достаточно.

GareevaFarida replied:
Да, я сначала так и собиралась сделать - использовать две константы. Но где-то вычитала, будто константы в прошлом и надо использовать перечисления. (Вполне возможно, что я переврала что-то и неверно поняла?) К сожалению, на источник сослаться не могу, потеряла :(

usharik commented on src/Exceptions/MyArrayDataException.java:
```diff
@@ -0,0 +1,10 @@
+package Exceptions;
+
+public class MyArrayDataException extends Exception {
+    public MyArrayDataException() {
```
Конструировать текст сообщения лучше здесь, используя переданные как параметры индексы;

GareevaFarida replied:
Я хотела собрать все ошибки по всему массиву, а не "спотыкаться" о первую.

GareevaFarida committed:
Перенесла формирование сообщения об ошибке в параметризованный конструктор исключения

GareevaFarida committed:
1. Перенесла формирование сообщения об ошибке в исключение.
2. Добавила условие на увеличение общей суммы только в случае отсутствия ошибок в данных массива.
